### ISSUE TITLE
Crash after about 10min in 

### ISSUE THREAD

**Issue by hoodie** [ebaauw/homebridge-hue#1062]
#### Issue

Hi there and happy Holidays,

I just set up a homebridge with homebridge-hue while at my parents but unfortunately it crashes after about 10min with this:

```bash
TypeError: Cannot read properties of undefined (reading 'id')
    at EventStreamClient.<anonymous> (/usr/local/lib/node_modules/homebridge-hue/lib/HueBridge.js:958:72)
    at EventStreamClient.emit (node:events:390:28)
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/homebridge-hue/lib/EventStreamClient.js:167:36)
    at IncomingMessage.emit (node:events:390:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at IncomingMessage.Readable.push (node:internal/streams/readable:228:10)
    at HTTPParser.parserOnBody (node:_http_common:141:24)
    at TLSSocket.socketOnData (node:_http_client:487:22)
    at TLSSocket.emit (node:events:390:28)
```

#### Log Messages
after which the entire homebridge needs to restart and I need to push the pairing button on the hue again.

```
[23/12/2021, 14:58:55] Got SIGTERM, shutting down Homebridge...
[23/12/2021, 14:59:00] [Hue] goodbye
[23/12/2021, 14:59:00] [HB Supervisor] Homebridge Process Ended. Code: 143, Signal: null
[23/12/2021, 14:59:05] [HB Supervisor] Restarting Homebridge...
[23/12/2021, 14:59:05] [HB Supervisor] Starting Homebridge with extra flags: -I
[23/12/2021, 14:59:06] [HB Supervisor] Started Homebridge v1.3.8 with PID: 6882
[23/12/2021, 14:59:09] Loaded config.json with 0 accessories and 2 platforms.
```

Any ideas about this? It's the latest release `0.13.31`

Thank you

**Comment by ebaauw** [comment]
The crash is caused by a bug in the error handler.  Not sure what's causing the error, though.  It looks like Homebridge Hue receives an event stream notification that it cannot parse.  Could you please capture and attach a a debug log file and attach the debug dump file? 

> after which the entire homebridge needs to restart

Best run child bridges to contain errors to a single plugin.

> I need to push the pairing button on the hue again

See the README and FAQ: you need to edit config.json, adding the obtained username (API key).

**Comment by hoodie** [comment]
thanks for the config.json tip!

here is my log from after a fresh restart

```log
[23/12/2021, 21:51:24] [homebridge-hue] This plugin is taking long time to load and preventing Homebridge from starting. See https://git.io/JtMGR for more info.
[23/12/2021, 21:51:24] [Hue] Philips hue: request 2: POST / {"devicetype":"homebridge-hue#homebridge"}
[23/12/2021, 21:51:24] [Hue] Philips hue: request 2: api error 101: link button not pressed
[23/12/2021, 21:51:24] [Hue] Philips hue: press link button on the bridge to create a user - retrying in 15s
[23/12/2021, 21:51:25] [Homebridge UI] Changes to config.json saved.
[23/12/2021, 21:51:29] [Homebridge UI] Changes to config.json saved.
[23/12/2021, 21:51:31] [Homebridge UI] Homebridge restart request received
[23/12/2021, 21:51:31] [Homebridge UI] UI / Bridge settings have not changed; only restarting Homebridge process
[23/12/2021, 21:51:31] [Homebridge UI] Sending SIGTERM to Homebridge
[23/12/2021, 21:51:31] Got SIGTERM, shutting down Homebridge...
[23/12/2021, 21:51:36] [Hue] goodbye
[23/12/2021, 21:51:36] [Hue] goodbye

[23/12/2021, 21:51:36] [HB Supervisor] Homebridge Process Ended. Code: 143, Signal: null
[23/12/2021, 21:51:41] [HB Supervisor] Restarting Homebridge...
[23/12/2021, 21:51:41] [HB Supervisor] Starting Homebridge with extra flags: -I
[23/12/2021, 21:51:41] [HB Supervisor] Started Homebridge v1.3.8 with PID: 1289
[23/12/2021, 21:51:44] Loaded config.json with 0 accessories and 2 platforms.
[23/12/2021, 21:51:45] Loaded 0 cached accessories from cachedAccessories.
[23/12/2021, 21:51:45] ---
[23/12/2021, 21:51:49] Loaded plugin: homebridge-config-ui-x@4.41.5
[23/12/2021, 21:51:49] Registering platform 'homebridge-config-ui-x.config'
[23/12/2021, 21:51:49] ---
[23/12/2021, 21:51:57] Loaded plugin: homebridge-hue@0.13.31
[23/12/2021, 21:51:57] Registering platform 'homebridge-hue.Hue'
[23/12/2021, 21:51:57] ---
[23/12/2021, 21:51:57] Loading 2 platforms...
[23/12/2021, 21:51:57] [Config] Initializing config platform...
[23/12/2021, 21:51:57] [Config] Running in Service Mode
[23/12/2021, 21:51:57] [Hue] Initializing Hue platform...
[23/12/2021, 21:51:57] [Hue] homebridge-hue v0.13.31, node v16.13.1, homebridge v1.3.8
[23/12/2021, 21:51:57] [Hue] searching bridges and gateways
[23/12/2021, 21:52:02] [Hue] Philips hue: Signify Netherlands B.V. BSB002 bridge ***********, api v1.48.0
[23/12/2021, 21:52:03] [Hue] Philips hue: 5 accessories
[23/12/2021, 21:52:03] [Hue] masked debug info dumped to /var/lib/homebridge/homebridge-hue.json.gz
[23/12/2021, 21:52:03] Initializing platform accessory 'Philips hue'...
[23/12/2021, 21:52:03] [Hue] Philips hue: 2 services
[23/12/2021, 21:52:03] Initializing platform accessory '****'...
[23/12/2021, 21:52:03] Initializing platform accessory '************'...
[23/12/2021, 21:52:03] Initializing platform accessory '**************'...
[23/12/2021, 21:52:03] Initializing platform accessory '*************'...
Setup Payload:
X-HM://*************
Enter this code with your HomeKit app on your iOS device to pair with Homebridge:
                       
    ┌────────────┐     
    │ ***-**-*** │     
    └────────────┘     
                       
[23/12/2021, 21:52:03] Homebridge v1.3.8 (HAP v0.9.7) (Homebridge 7A3D) is running on port 51250.
[23/12/2021, 21:52:03] [Hue] Philips hue: event stream connected to https://**************/eventstream/clip/v2
[23/12/2021, 21:59:12] TypeError: Cannot read properties of undefined (reading 'id')
    at EventStreamClient.<anonymous> (/usr/local/lib/node_modules/homebridge-hue/lib/HueBridge.js:958:72)
    at EventStreamClient.emit (node:events:390:28)
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/homebridge-hue/lib/EventStreamClient.js:167:36)
    at IncomingMessage.emit (node:events:390:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at IncomingMessage.Readable.push (node:internal/streams/readable:228:10)
    at HTTPParser.parserOnBody (node:_http_common:141:24)
    at TLSSocket.socketOnData (node:_http_client:487:22)
    at TLSSocket.emit (node:events:390:28)
[23/12/2021, 21:59:12] Got SIGTERM, shutting down Homebridge...
[23/12/2021, 21:59:18] [Hue] goodbye
[23/12/2021, 21:59:18] [HB Supervisor] Homebridge Process Ended. Code: 143, Signal: null
[23/12/2021, 21:59:23] [HB Supervisor] Restarting Homebridge...
[23/12/2021, 21:59:23] [HB Supervisor] Starting Homebridge with extra flags: -I
[23/12/2021, 21:59:23] [HB Supervisor] Started Homebridge v1.3.8 with PID: 1571
[23/12/2021, 21:59:26] Loaded config.json with 0 accessories and 2 platforms.
[23/12/2021, 21:59:26] Loaded 0 cached accessories from cachedAccessories.
[23/12/2021, 21:59:26] ---
```

**Comment by ebaauw** [comment]
>here is my log from after a fresh restart

Thanks.  I'm afraid you didn't enable Homebridge debug mode?  Now it doesn't log the notification.  Next time, better attach the file, instead of copy/pasting large amounts of text.

> and the dump

Only four OSRAM plugs?  I wouldn't expect any "weird" notifications from them.  Maybe the zone - I don't have any of those.  Were you doing any configuration in the Hue app, when the crash occurred?

**Comment by hoodie** [comment]
merry xmas,
I'm keeping an eye on it, but unfortunately it has not reoccurred since I turned on debug mode

**Comment by ebaauw** [comment]
Error handler fix in v0.13.32.
